### ISSUE TITLE
Cycle button: list of action or grouped actions

### ISSUE THREAD

**Issue by ArthurAK** [bitfocus/companion#2286]
### Is this a feature relevant to companion itself, and not a module?

- [X] I believe this to be a feature for companion, not a module

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the feature

Method # 1: List of actions (Preferred method, but more complicated to create): 

Create List  (Edit List) > 
Name: [] 
Dropdown: + Add Action 
-- keep adding actions in order

![brave_MNXR4VIPud](https://user-images.githubusercontent.com/44576346/212782140-de118fe2-cc5e-4dff-ae77-87b868221e30.png)

Variable type of page to create a lists of actions. Then a button that will cycle thru that list
Button: Action: Internal: List
Dropdown: [Select List by ID / Name]
Dropdown: Direction: [Next, Previous, Set] (looped)
Dropdown: [1-N]
![image](https://user-images.githubusercontent.com/44576346/212780888-6cdaa3de-8a22-4f5f-b3d3-39c82899e45e.png)

Extra internal control over button with the list: another button to select or cycle lists into a button:
Action: Internal: Button Action
Dropdown: Page: Y Button Z (similar to button press&release)
Dropdown: next / previous / set
Dropdown: [List1, List2 ... ListN]

Method # 2: Grouped actions
Similar check box than latch / toggle, but "Groups / Cycles"
(Extra: dropdown for selecting direction)
Instead of latch / unlatch, there will be "Group 1, Group 2 ... Group N, with a button to remove or add a group.
Each group has "+ Add Action" and all actions will be exected when button is pressed (Either no delays allowed or next button press will abort actions).

Extra: internal: button: cycle, Next/previous to call this list and it's position

Curently there is workaround, that consists of using many pages to create multiple toggle buttons and then using aborts etc. to select thru them, but this is tidious/labor and brain intensive with little copy/paste ability and waste of pages&buttons.

### Usecases

Creating an input selector for up/downstreamkey inputs, or cycling thru media in VLC / Playoutbee. Or cycling thru inputs on Multiview or Supersource boxes.

**Comment by MeestorX** [comment]
Not really sure what you're looking to achieve (it's hard to follow what you've written), but if it's adding steps to buttons, then look at v3, Julian spent a lot of time on that and it's pretty dialled in!

**Comment by ArthurAK** [comment]
> Not really sure what you're looking to achieve (it's hard to follow what you've written), but if it's adding steps to buttons, then look at v3, Julian spent a lot of time on that and it's pretty dialled in!

Sorry, I pressed enter accidentally mid writing, didn't have the method 2 written. Is there documentation on how it works? I don't prefer to update to v3 yet, as not stable.

**Comment by Julusian** [comment]
the documentation hasnt been updated for v3, so this isnt really documented. I was going to provide some links to PRs, but I think this feature has been through too many revisions for the PRs to be coherent anymore.

The basics is that each button now has multiple 'steps'. 
![image](https://user-images.githubusercontent.com/1327476/212787123-6bbe7522-3d63-44aa-9bcc-de91f1070d7f.png)
These each have their own press and release actions. By default upon each release, it will set itself to the next step, so if you keep pressing it it will run step 1, 2, 3 etc. Or you can turn that off, and define actions yourself on when to change the step of a button.

This sounds similar to what you are asking for in method 2 (and that sounds a lot closer to an earlier implementation of steps, before it got expanded upon)

As for stability, depending on the modules you use, 3.0 could be perfectly stable. Its pretty much just modules which need some stability work now, which is also generally that many have been broken and will consistently not work. I have been running 3.0 in my office environment since I think July or something, and apart from a couple of times where I broke something and it wouldn't even launch, it has been running perfectly. I hesitate to call it fully stable, but it can be stable, depending on what you use in it

**Comment by ArthurAK** [comment]
> the documentation hasnt been updated for v3, so this isnt really documented. I was going to provide some links to PRs, but I think this feature has been through too many revisions for the PRs to be coherent anymore.
> 
> The basics is that each button now has multiple 'steps'. ![image](https://user-images.githubusercontent.com/1327476/212787123-6bbe7522-3d63-44aa-9bcc-de91f1070d7f.png) These each have their own press and release actions. By default upon each release, it will set itself to the next step, so if you keep pressing it it will run step 1, 2, 3 etc. Or you can turn that off, and define actions yourself on when to change the step of a button.
>

Seems to be just a great implementation!

> 
> This sounds similar to what you are asking for in method 2 (and that sounds a lot closer to an earlier implementation of steps, before it got expanded upon)
> 
> As for stability, depending on the modules you use, 3.0 could be perfectly stable. Its pretty much just modules which need some stability work now, which is also generally that many have been broken and will consistently not work. I have been running 3.0 in my office environment since I think July or something, and apart from a couple of times where I broke something and it wouldn't even launch, it has been running perfectly. I hesitate to call it fully stable, but it can be stable, depending on what you use in it

I use Behringer XAir, BM Atem Mini, HTTP, OSC, H2R Graphics (which I think is not yet done at all for v3), Vicreo, and maybe something I can't remember but have it installed.

**Comment by Julusian** [comment]
As it sounds like this is already done in 3.0/beta, Im going to close this
